tests/profile.test.js

```diff
@@ -385,4 +401,202 @@ describe('Profile and accounts workflow:', function () {
     browser.url('http://localhost:3000')
     browser.waitForVisible('#foundKeystore #btn-foundKeystore-login')
   })
+
+  describe('password reset', () => {
+    it('should not allow the user to change their password if they enter the incorrect current password', function () {
+      createUserAndLogin(browser)
+      browser.url('http://localhost:3000/profile')
+      browser.waitForClickable('#edit-profile')
+      browser.pause(500)
```

Comment:
browser.pause() is EVIL in tests - this will lead to unpredictable behavior (especially in circleci, where we cannot count on any timing at all....) - so all of these need to go.

The reason why you need the browser.pause here is because you used the bootstrap Modal machinery to open the editprofile menu, which messes up the logic of opening/closig modals (apparently - I don't understand the exact mechanics, but it seems there is some open-close-reopen dynamcs going on). I'd suggest removign the `Modal.show` from the profile.js when you click on edit-profile, and just show/hide a dom element there (justa sw e do with the alerts).

Reply:
Gotcha, going to give this a try now

Reply:
My only concern with this menu no longer being a modal is that we will lose out on the graying out of the background/auto-dismissing of the menu when clicking elsewhere, at least out of the box. 

Will have to check w/ @pedrocasa once the new way is implemented

Reply:
@jellegerbrandy I refactored the edit profile menu so that it is not using bootstrap modal. Here is the experience now (cc @pedrocasa):

![profilemenu](https://user-images.githubusercontent.com/7697924/32867816-49a9fe68-ca3d-11e7-92a1-357da0de3c82.gif)

However, the system that was in place previously _did_ work reliably from the user's perspective, but just not reliably for tests. I am a bit weary of changing the implementation just to satisfy tests (despite the fact that I am a huge testing advocate), especially at the expense of UX. Did you envision a way to implement this alternative solution while keeping all of the same styles/interactions as the main modal?

Reply:
Yes, that is a good point (about not wanting to change to code just for getting the tests to work) - I agree. 
In this case, it is also a matter of code organization - we use the bootstrap modal machinery in the one-modal-at-a-time mode for showing, well, the modal dialgo boxes. So i think it makes sense to use the boostrap `show()` and `hide()` stuff only for the modal dialogs - not just for the tests, but also to have a predictable API to program against.